tools/dotnet-linker/Steps/ManagedRegistrarStep.cs

```diff
@@ -0,0 +1,1160 @@
+						if (delegateProxyType is null) {
+							exceptions.Add (ErrorHelper.CreateWarning (App, 4176, method, Errors.MT4176 /* "Unable to locate the delegate to block conversion type for the return value of the method {0}." */, method.FullName));
+						} else {
+							createBlockMethod = StaticRegistrar.GetCreateBlockMethod (delegateProxyType);
+							if (createBlockMethod is null) {
+								delegateProxyField = delegateProxyType.Fields.SingleOrDefault (v => v.Name == "Handler");
+								if (delegateProxyField is null) {
+									AddException (ErrorHelper.CreateWarning (99, "No delegate proxy field on {0}", delegateProxyType.FullName));
+								}
+							}
+						}
+					}
+
+					// the delegate is already on the stack
+					if (createBlockMethod is not null) {
+						EnsureVisible (method, createBlockMethod);
+						il.Emit (OpCodes.Call, method.Module.ImportReference (createBlockMethod));
+						il.Emit (OpCodes.Call, abr.RegistrarHelper_GetBlockPointer);
+					} else if (delegateProxyField is not null) {
+						EnsureVisible (method, delegateProxyField);
+						il.Emit (OpCodes.Ldsfld, method.Module.ImportReference (delegateProxyField));
+						il.Emit (OpCodes.Ldstr, signature);
+						il.Emit (OpCodes.Call, abr.BlockLiteral_CreateBlockForDelegate);
+					} else {
+						il.Emit (OpCodes.Ldtoken, method);
+						il.Emit (OpCodes.Call, abr.RegistrarHelper_GetBlockForDelegate);
+					}
+				}
+				nativeType = abr.System_IntPtr;
+				return true;
+			}
+
+			AddException (ErrorHelper.CreateError (99, "Don't know how (1) to convert {0} between managed and native code: {1}. Method: {2}", type.FullName, type.GetType ().FullName, GetMethodSignatureWithSourceCode (method)));
+			return false;
+		}
+
+		void EnsureVisible (MethodDefinition caller, FieldDefinition field)
+		{
+			field.IsPublic = true;
+			EnsureVisible (caller, field.DeclaringType);
+		}
+
+		void EnsureVisible (MethodDefinition caller, TypeDefinition type)
+		{
+			if (type.IsNested) {
+				type.IsNestedPublic = true;
+				EnsureVisible (caller, type.DeclaringType);
+			} else {
+				type.IsPublic = true;
+			}
+		}
+
+		void EnsureVisible (MethodDefinition caller, MethodReference method)
+		{
+			var md = method.Resolve ();
+			md.IsPublic = true;
+			EnsureVisible (caller, md.DeclaringType);
+		}
+
+		bool IsOutParameter (MethodDefinition method, int parameter, ParameterDefinition baseParameter)
+		{
+			return method.Parameters [parameter].IsOut || baseParameter.IsOut;
+		}
+
+		StaticRegistrar StaticRegistrar {
+			get { return DerivedLinkContext.StaticRegistrar; }
+		}
+
+		CustomAttribute CreateUnmanagedCallersAttribute (string entryPoint)
+		{
+			var unmanagedCallersAttribute = new CustomAttribute (abr.UnmanagedCallersOnlyAttribute_Constructor);
+			unmanagedCallersAttribute.Fields.Add (new CustomAttributeNamedArgument ("EntryPoint", new CustomAttributeArgument (abr.System_String, "_" + entryPoint)));
```

Comment:
Is there any particular reason to prefix the entrypoint name with `_`? NativeAOT already adds the `_` prefix by itself on macOS/iOS-like platforms, and I would expect other runtimes to behave the same.

(It shouldn't necessarily matter as long as it's consistent between the managed and native code.)

Reply:
I had to add it otherwise it didn't work (lots of unresolved symbols). Then again I was working with Mono's AOT compiler, so maybe that's a bug that will have to be fixed there.

Reply:
Actually it looks like Ivan fixed it already for .NET 8: https://github.com/dotnet/runtime/issues/79491

Reply:
In other words: the underscore is correct for .NET 7, but not for .NET 8.

Reply:
Fixed!

Reply:
Confirmed to be fixed on net8.0 with the latest change! Thanks.